### PR TITLE
feat: support electron V5 

### PR DESCRIPTION
**Pull request by magne4000**
Diff description
--

- Upgraded Electron, Process Manager and Mocha dependencies
- Changed calls for schema registration (https://github.com/electron/electron/blob/master/docs/api/breaking-changes.md#privileged-schemes-registration)
- Changed call for isolated worlds (https://github.com/electron/electron/blob/master/docs/api/breaking-changes.md#webframe-isolated-world-apis)
- Changed playground browser window options web preferences (https://github.com/electron/electron/blob/master/docs/api/breaking-changes.md#new-browserwindow-webpreferences-)
- Reverted schema protocol handler on hostname binding
- Added a workaround for the missing `window.opener` reference with native window open (https://github.com/electron/electron/issues/18032)

### CODE REVIEW

**Comment by hugomano** [comment]
Origin `chrome-extension` is now a secure origin in the renderer, thank you to https://github.com/getstation/electron-chrome-extension/pull/47

**Comment by hugomano** [comment]
https://github.com/electron/electron/issues/18032

**Comment by hugomano** [comment]
to test
- [ ] test with a browser window instead of webview 
- [ ] test with no new guest
- [ ] with navigation

File: src/renderer/window-setup.js [hunk 1]
```diff
@@ -99,23 +99,46 @@ module.exports = (win, ipcRenderer, guestInstanceId, openerId, nonNativeWinOpenF
...
+    // we lost the opener reference.
+    //
+    // We polyfill the *postMessage* method to forward the message via the main
+    // (src/browser/engine/window.ts). The main broadcast the message for
+    // all allowed webContents (matching type, protocol) only if the
+    // webContents sender match the *targetOrigin*
+    //
+    // used by: Mixmax and other apps using Google OAuth
+    // issue ref: https://github.com/electron/electron/issues/18032
+    win.opener = {
```
Comment (alexstrat) [inline src/renderer/window-setup.js]: After the fix of electron/electron#18032 in electron 5.0.4, I think we can remove most of these workarounds, no?

Comment (hugomano) [reply src/renderer/window-setup.js]: yep I on it

Comment (alexstrat) [reply src/renderer/window-setup.js]: 🆙

**Pull request merged**